### PR TITLE
Fix an error for `RSpec/Rails/HaveHttpStatus` with comparison with strings containing non-numeric characters

### PR DESCRIPTION
**Pull request by ydah**
Fix: https://github.com/rubocop/rubocop-rspec/issues/1623

______________________________________________________________________

Before submitting the PR make sure the following are checked:

- [x] Feature branch is up-to-date with `master` (if not - rebase it).
- [x] Squashed related commits together.
- [x] Added tests.
- [-] Updated documentation.
- [x] Added an entry to the `CHANGELOG.md` if the new code introduces user-observable changes.
- [x] The build (`bundle exec rake`) passes (be sure to run this locally, since it may produce updated documentation that you will need to commit).

### CODE REVIEW

**Comment by pirj** [comment]
The original reported issue is incorrect. I wouldn't bend our code to tolerate that.
Still, we could work around the formatting issue by using `have_http_status(%{status})` instead of `have_http_status(%<status>i)`, see https://github.com/rubocop/rubocop/issues/8827.

File: lib/rubocop/cop/rspec/rails/have_http_status.rb [hunk 1]
```diff
@@ -37,12 +37,14 @@ class HaveHttpStatus < ::RuboCop::Cop::Base
 
           def on_send(node)
             match_status(node) do |response_status, to, match, status|
+              return unless status.to_s.match?(/\A\d+\z/)
```
Comment (pirj) [inline lib/rubocop/cop/rspec/rails/have_http_status.rb:40]: Nitpick: “404 “ with a trailing space would be ignored. Should it?

**Review by pirj — APPROVED** [review]
Thank you!

File: lib/rubocop/cop/rspec/rails/have_http_status.rb (hunk 1 above)
Comment (bquorning) [reply lib/rubocop/cop/rspec/rails/have_http_status.rb:40]: I think that is an acceptable edge case.

**Review by bquorning — APPROVED** [review]

**Pull request merged**